File: whdon_app/src/main/java/nl/whitedove/washetdroogofniet/StatsWindComparator.java
package nl.whitedove.washetdroogofniet;

import java.util.Comparator;

public class StatsWindComparator implements Comparator<StatistiekWind> {
    static final StatsWindComparator instance=new StatsWindComparator();

    public int compare(StatistiekWind left, StatistiekWind right) {
        return Double.compare(left.getWindDir().getValue(), right.getWindDir().getValue());
    }
}

File: app/src/main/java/nl/whitedove/washetdroogofniet/StatsWindComparator.kt
package nl.whitedove.washetdroogofniet

import java.util.Comparator

class StatsWindComparator : Comparator<StatistiekWind> {

    override fun compare(left: StatistiekWind, right: StatistiekWind): Int {
        return java.lang.Double.compare(left.windDir.value.toDouble(), right.windDir.value.toDouble())
    }

    companion object {
        internal val instance = StatsWindComparator()
    }
}
